### PR TITLE
PIMP-66-Calendar-Events

### PR DESCRIPTION
**Pull request by DeJona**

### CODE REVIEW

File: src/app/calendar/event-editor/calendar-event-editor.component.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  constructor(private calendarService: CalendarService) {
+
+  }
+
+  showDialog(calendarEvent?: CalEvent) {
+    this.eventStart = null;
+    this.eventEnd = null;
+    if(calendarEvent) {
+      this.event = this.clone(calendarEvent); // avoid UI changes if not saved
```
Comment (haffla) [inline src/app/calendar/event-editor/calendar-event-editor.component.ts]: In ES6 (and TypeScript) you can do `this.event = Object.assign({}, calendarEvent);` to clone the event.

File: src/app/services/calendar.service.ts [hunk 2]
```diff
@@ -80,4 +82,48 @@ export default class CalendarService {
...
+    ).subscribe(
+      response => {
+        this.events = this.events.filter(evt => evt.key !== event.key);
+        this.eventsChange.next(this.events);
+      },
+      err => console.log(err)
+    )
+  }
+
+  private mapCalendarEvent(event: CalEvent): any {
```
Comment (haffla) [inline src/app/services/calendar.service.ts]: So this function can now be used up in `mapCalendarEvents()`, no? (lines ~39-45)

File: package.json [hunk 3]
```diff
@@ -21,13 +21,16 @@
     "@angular/platform-browser": "~2.1.0",
     "@angular/platform-browser-dynamic": "~2.1.0",
     "@angular/router": "~3.1.0",
+    "angular-calendar": "~0.4.0",
     "core-js": "^2.4.1",
+    "font-awesome": "^4.7.0",
```
Comment (haffla) [inline package.json]: Font awesome was already in the project. But your way of including it is better. Please remove it from index.html.

**Comment by haffla** [comment]
You need to declare the new CalendarEventEditor component in the spec to make the tests green

**Comment by DeJona** [comment]
@haffla Thanks for the feedback.

File: src/app/services/calendar.service.ts (hunk 2 above)
Comment (DeJona) [inline src/app/services/calendar.service.ts]: Better not, `mapCalendarEvent()` maps from frontend to backend, `mapCalendarEvents()` the other way round.

**Pull request merged**